CHANGELOG.md

```diff
@@ -3,6 +3,344 @@
+let channel = guild.create_channel(&http, builder).await?;
+```
+
+Note that in this particular example, the channel name is now a mandatory field that must be passed in when constructing the builder. Mutating the builder with subsequent calls to `CreateChannel::name` will change the underlying value. Additionally, all methods on builders now take `mut self` and return `Self`, instead of taking and returning `&mut self`/`&mut Self`. This allows for inline initialization as in the second example above. Also, builders no longer wrap a `pub HashMap<&'static str, T>`; the hashmap has been flattened into concrete private fields.
+
+Some benefits to this new approach to builders are:
+
+ 1. Because every closure has a type unique to it, each call to a method taking a closure would be monomorphized separately, resulting in binary bloat. This is no longer the case.
+ 2. Builders can be constructed once and then cloned and re-used multiple times.
+ 3. Enforcement of field requirements (as dictated by the Discord API docs) prevents invalid requests at compile-time.
+
+### Attachments
+
+* The `AttachmentType` enum has been replaced with a `CreateAttachment` builder struct. This struct has the `file`, `path`, and `url` constructors that eagerly evaluate the data passed to them - `CreateAttachment` simply stores the resulting raw data. This is in contrast to `AttachmentType` which lazily carried filepaths/urls with it, and had `data` and `filename` methods for resolving them.
+* The `utils::encode_image` function has been replaced with `CreateAttachment::to_base64`.
+
+### Collectors
+
+Collectors have been redesigned and simplified at no cost to expressibility. There is now a generic `collector::collect` function which takes a closure as argument, letting you filter events as they stream in.
+* The specific collectors (`ComponentInteractionCollector`, `ModalInteractionCollector`, `MessageCollector`, and `ReactionCollector`) are simply convenience structs that wrap this underlying function.
+* `EventCollector` is now deprecated, as its use usually involved anti-patterns around fallibility. However, its functionality can still be replicated using `collector::collect`. See example 10 for more details.
+* The `RelatedId` and `RelatedIdsForEventType` types have been removed as they were only used by `EventCollector`. Methods for retrieving them from events have also been removed; if users wish to extract "related" ids from an event, they should do so directly from the event's fields. The removed methods are the following:
+    - `Event::user_id`
+    - `Event::guild_id`
+    - `Event::channel_id`
+    - `Event::message_id`
+    - `EventType::related_ids`
+
+### Commands
+
+Serenity now uses simply `command` instead of `application_command` in all places, except for the `Permissions::USE_APPLICATION_COMMANDS` constant. This includes methods on the `Http`, `GuildId`, `Guild`, `PartialGuild`, and `Command` types, as well as a few miscellaneous places:
```

Comment:
```suggestion
Serenity now uses simply `command` instead of `application_command` in all places to make paths less unwieldy, except for the `Permissions::USE_APPLICATION_COMMANDS` constant. This includes methods on the `Http`, `GuildId`, `Guild`, `PartialGuild`, and `Command` types, as well as a few miscellaneous places:
```